File: Data Structures & Algorithms/jump-game/submission-1.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        n = len(nums)
        b = [False] * n
        b[0] = True
        for i in range(n - 1):
            if b[i]:
                for j in range(nums[i]+1):
                    if i + j < n:
                        if i + j == n - 1:
                            return True
                        b[i + j] = True
        return b[-1]

File: Data Structures & Algorithms/duplicate-integer/submission-1.swift
class Solution {
    func hasDuplicate(_ nums: [Int]) -> Bool {
        var m: [Int: Int] = [:]
        for num in nums {
            if m[num] == nil {
                m[num] = 0
            }
            m[num] = m[num]! + 1
            if m[num]! > 1 {
                return true
            }
        }
        return false
    }
}
